hack/ci/generate-junit-xml-report.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+# Script to generate a JUnit XML report from a JSON go test output.
+#
+
+set -euo pipefail
+
+main() {
+    local input_file=${1:-"input file"}
+    local xml_report=${input_file%.*}.xml
+
+    # exit without error when there is no input file
+    if [[ ! -f $input_file ]]; then
+        echo "No $input_file to generate a JUnit XML report."
+        exit 0
+    fi
+  
+    # temporary filter out lines containing a space in the timestamp,
+    # see https://github.com/elastic/cloud-on-k8s/issues/3560.
+    gotestsum \
+        --junitfile "$xml_report" \
+        --raw-command grep -v '"Time":"[^"]*\s[^"]*"' "$input_file" || \
```

Comment:
I'm wondering if it's worth parsing the whole timestamp, maybe a bit less readable but it would also prevent some other kinds of timestamp "corruption":

```suggestion
        --raw-command grep -P '"Time":"(?:[1-9]\d{3}-(?:(?:0[1-9]|1[0-2])-(?:0[1-9]|1\d|2[0-8])|(?:0[13-9]|1[0-2])-(?:29|30)|(?:0[13578]|1[02])-31)|(?:[1-9]\d(?:0[48]|[2468][048]|[13579][26])|(?:[2468][048]|[13579][26])00)-02-29)T(?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d(?:\.\d{1,9})?(?:Z|[+-][01]\d:[1-5]\d)"' "$input_file" || \
```

Reply:
Yes but I think it would be interesting to detect different timestamp corruption for now. So I'm in favor of keeping the regexp simpler.